src/processor/ValueSetProcessor.ts

```diff
@@ -46,13 +44,35 @@ export class ValueSetProcessor {
       ...CaretValueRuleExtractor.processResource(input, fisher, input.resourceType, config)
     );
     if (input.compose) {
-      input.compose.include?.forEach((vsComponent: any) => {
+      input.compose.include?.forEach((vsComponent: fhirtypes.ValueSetComposeIncludeOrExclude) => {
         newRules.push(ValueSetFilterComponentRuleExtractor.process(vsComponent, input, true));
         newRules.push(ValueSetConceptComponentRuleExtractor.process(vsComponent, true));
+        vsComponent.concept?.forEach(includedConcept => {
+          newRules.push(
+            ...CaretValueRuleExtractor.processConcept(
+              includedConcept,
+              [includedConcept.code],
+              target.name,
+              'ValueSet',
+              fisher
+            )
+          );
+        });
       });
-      input.compose.exclude?.forEach((vsComponent: any) => {
+      input.compose.exclude?.forEach((vsComponent: fhirtypes.ValueSetComposeIncludeOrExclude) => {
         newRules.push(ValueSetFilterComponentRuleExtractor.process(vsComponent, input, false));
         newRules.push(ValueSetConceptComponentRuleExtractor.process(vsComponent, false));
+        vsComponent.concept?.forEach(excludedConcept => {
```

Comment:
The same comment applies here re: having caret rules follow their corresponding concept rule rather than clumping all concept rules together and then all caret rules together.